src/io/flutter/view/WidgetPreviewUrlProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2025 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.view;
+
+import io.flutter.devtools.DevToolsUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class WidgetPreviewUrlProvider implements BrowserUrlProvider {
+  private final @NotNull String url;
+  private boolean isBackgroundBright;
+
+  public WidgetPreviewUrlProvider(@NotNull String url, boolean isBackgroundBright) {
+    this.url = url;
+    this.isBackgroundBright = isBackgroundBright;
+  }
+
+  @Override
+  public void setWidgetId(@NotNull String widgetId) {
+
+  }
+
+  @Override
+  public @NotNull String getBrowserUrl() {
+    return url + "?theme=" + (isBackgroundBright ? "light" : "dark");
```

Comment:
Do we support background and foreground colors for theming DevTools? If so, we should also pass those values here.